{lang}[iimkl/2025b] SciPy-bundle v2025.07

branfosj opened the pull request:
(created using `eb --new-pr`)

* Requires fix in https://github.com/easybuilders/easybuild-framework/pull/4981

branfosj commented:
@boegelbot please test @ jsc-zen3

schiotz commented:
Test report by @schiotz
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
svol.fysik.dtu.dk - Linux Rocky Linux 8.10, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 3.6.8
See https://gist.github.com/schiotz/cd788c261b93803944e3c1800f517e95 for a full test report.

branfosj commented:
Test report by @branfosj
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
bear-pg0105u03a - Linux RHEL 8.10, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/f3c10d48f857d0bb95f7b2509ee32ba3 for a full test report.

branfosj commented:
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0208u17a - Linux RHEL 8.10, x86_64, AMD EPYC 9554 64-Core Processor (zen4), Python 3.6.8
See https://gist.github.com/branfosj/08c57f09f47230ce8a3cff48246d885a for a full test report.

branfosj commented:
@boegelbot please test @ jsc-zen3

branfosj commented:
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0201u03a - Linux RHEL 8.10, x86_64, INTEL(R) XEON(R) PLATINUM 8570 (sapphirerapids), Python 3.6.8
See https://gist.github.com/branfosj/bfbad3c09c99682f3a5929f738649137 for a full test report.

branfosj commented:
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0208u17a - Linux RHEL 8.10, x86_64, AMD EPYC 9554 64-Core Processor (zen4), Python 3.6.8
See https://gist.github.com/branfosj/617f2a7823e641ea6b0cc3b07369ddef for a full test report.

branfosj commented:
> @branfosj From the support ticket I filed:
> 
> > The recommended workaround for now is to disable this particular pass by adding '-mllvm -disable-hir-backward-temp-dependencies-resolver'.

I've added that and removed the patch. We might want to target it more than this, but it will be good to check that it works.

branfosj commented:
@boegelbot please test @ jsc-zen3

branfosj commented:
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0201u03a - Linux RHEL 8.10, x86_64, INTEL(R) XEON(R) PLATINUM 8570 (sapphirerapids), Python 3.6.8
See https://gist.github.com/branfosj/7297785440adfb4da15bfb3abc09a4c1 for a full test report.

branfosj commented:
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0105u03a - Linux RHEL 8.10, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/95f902e19159a76322f0e11884293896 for a full test report.

branfosj commented:
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0208u17a - Linux RHEL 8.10, x86_64, AMD EPYC 9554 64-Core Processor (zen4), Python 3.6.8
See https://gist.github.com/branfosj/96e63baf1327074fc6e83c3cf37daff9 for a full test report.

boegel commented:
Test report by @boegel
**FAILED**
Build succeeded for 3 out of 4 (1 easyconfigs in total)
node3590.doduo.os - Linux RHEL 9.4, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.9.18
See https://gist.github.com/boegel/1d8ea313f0c45bbe03755a3db4909dbc for a full test report.

Thyre commented:
> Test report by @boegel
**FAILED**
Build succeeded for 3 out of 4 (1 easyconfigs in total)
node3590.doduo.os - Linux RHEL 9.4, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.9.18
See https://gist.github.com/boegel/1d8ea313f0c45bbe03755a3db4909dbc for a full test report.

We require the framework fixes for the rpath wrapper, otherwise Meson fails to detect lots of features unfortunately.
I think that commit used is a bit too old for that (Tue Aug 19 17:50:45 2025 +0200)

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3590.doduo.os - Linux RHEL 9.4, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.9.18
See https://gist.github.com/boegel/1ed33253a98d9f48320c32aa39eaaf2a for a full test report.

boegel commented on easybuild/easyconfigs/s/SciPy-bundle/SciPy-bundle-2025.07-iimkl-2025b.eb:
```diff
@@ -8,6 +8,10 @@ description = "Bundle of Python packages for scientific software"
 toolchain = {'name': 'iimkl', 'version': '2025b'}
 toolchainopts = {'pic': True, 'lowopt': True, 'strict': True,
+                 # '-fveclib=none' to avoid 'undefined symbol: pow8_h' in numpy
+                 # see https://github.com/numpy/numpy/issues/27111
```
@branfosj Thoughts on this suggestion?
